equinox/_tree.py

```diff
@@ -406,15 +418,15 @@ def is_leaf(node):
 
 
 def tree_check(pytree: Any) -> None:
-    """Checks if the PyTree is well-formed: does it have no self-references, and does
-    it have no duplicate layers.
-
-    Precisely, a "duplicate layer" is any PyTree node with at least one child node.
-
-    !!! info
+    """Checks if the PyTree has no self-references, and if all non-leaf nodes unique
+    Python objects. (For example something like `x = [1]; y = [x, x]` would fail as `x`
```

Comment:
```suggestion
    """Checks if the PyTree has no self-references, and if all non-leaf nodes are unique
    Python objects. (For example something like `x = [1]; y = [x, x]` would fail as `x`
```

I think you missed a word here, no?